core/server/common/src/main/java/alluxio/master/MasterProcess.java

```diff
@@ -187,9 +188,32 @@ void registerServices(GrpcServerBuilder serverBuilder,
    * @return whether the grpc server became ready before the specified timeout
    */
   public boolean waitForGrpcServerReady(int timeoutMs) {
+    return pollFor(this + " to start", this::isGrpcServing, timeoutMs);
```

Comment:
the isGrpcServing function has data races on mGrpcServer and mStarted in GrpcServer.java. These need to be volatile or atomic boolean/pointer

Reply:
made appropriate variables volatile.